### ISSUE TITLE
4.0 doesn't compile on illumos-based systems

### ISSUE THREAD

**Issue by jeffpc** [PowerDNS/pdns#3028]
Trying to compile the latest and greatest pdns on my OpenIndiana Hipster (an illumos-based distro), I get the following errors:

```
$ CXXFLAGS=-std=c++11 gmake

PLEASE READ: If you get an error mentioning #include <boost/something.hpp>, please read README
PLEASE READ: for an easy fix!

g++ -std=c++11 -Iext/rapidjson/include -I/tmp/pdns-recursor-0.0.453g1247522/ext/mbedtls/include -Wall    -O3   -pthread -Iext/yahttp -DHAVE_CONFIG_H -D__MAKECONTEXT_V2_SOURCE -DHAVE_STRCASESTR -DSYSCONFDIR='"/etc/powerdns/"' -DLOCALSTATEDIR='"/var/run/"' -c portsmplexer.cc -o optional/portsmplexer.o  ; true
In file included from sstuff.hh:7:0,
                 from portsmplexer.cc:10:
iputils.hh: In member function ‘NetmaskTree<T>::node_type& NetmaskTree<T>::insert(const key_type&)’:
iputils.hh:493:73: error: there are no arguments to ‘be32toh’ that depend on a template parameter, so a declaration of ‘be32toh’ must be available [-fpermissive]
       std::bitset<32> addr(be32toh(key.getNetwork().sin4.sin_addr.s_addr));
                                                                         ^
iputils.hh:493:73: note: (if you use ‘-fpermissive’, G++ will accept your code, but allowing the use of an undeclared name is deprecated)
iputils.hh:512:47: error: there are no arguments to ‘be64toh’ that depend on a template parameter, so a declaration of ‘be64toh’ must be available [-fpermissive]
       std::bitset<64> addr_low(be64toh(addr[1]));
                                               ^
iputils.hh:513:48: error: there are no arguments to ‘be64toh’ that depend on a template parameter, so a declaration of ‘be64toh’ must be available [-fpermissive]
       std::bitset<64> addr_high(be64toh(addr[0]));
                                                ^
iputils.hh: In member function ‘const node_type* NetmaskTree<T>::lookup(const ComboAddress&, int) const’:
iputils.hh:571:62: error: there are no arguments to ‘be32toh’ that depend on a template parameter, so a declaration of ‘be32toh’ must be available [-fpermissive]
       std::bitset<32> addr(be32toh(value.sin4.sin_addr.s_addr));
                                                              ^
...
```

It turns out that `be32toh` and friends don't exist on illumos:

```
$ grep -R be32toh /usr/include
grep: warning: /usr/include/gphoto2/gphoto2: recursive directory loop
```

Illumos does have `sys/byteorder.h` with `ntohl` & friends: http://src.illumos.org/source/xref/illumos-gate/usr/src/uts/common/sys/byteorder.h

**Comment by cmouse** [comment]
Maybe they are under some covert name? Can you look into /usr/include/endian.h

**Comment by cmouse** [comment]
The relevant macros are in http://src.illumos.org/source/xref/illumos-gate/usr/src/uts/common/sys/byteorder.h

**Comment by jeffpc** [comment]
Tested with 439b8c5842016335474a10bb5b62c73b8fc53348 + fix for #3239 on OpenIndiana Hipster - everything seems to build just fine.
